cf-agent/cf-agent.c

```diff
@@ -308,6 +310,11 @@ static GenericAgentConfig *CheckOpts(int argc, char **argv)
             TIMING = true;
             break;
 
+        case 'w':
+            Log(LOG_LEVEL_INFO, "Setting workdir to '%s'", optarg);
+            setenv("CFENGINE_TEST_OVERRIDE_WORKDIR", optarg, 1);
```

Comment:
`setenv` is not a POSIX function (can you believe it??). You'll have to use `putenv`, which will require you to allocate the memory you give to it. This will leak if you later give it a different string, but I think we don't need to worry about that. This will happen only here, and only once.